velox/dwio/dwrf/writer/ColumnWriter.cpp

```diff
@@ -2036,9 +2149,20 @@ std::unique_ptr<BaseColumnWriter> BaseColumnWriter::create(
     case TypeKind::INTEGER:
       return std::make_unique<IntegerColumnWriter<int32_t>>(
           context, type, sequence, onRecordPosition);
-    case TypeKind::BIGINT:
+    case TypeKind::BIGINT: {
+      if (format == DwrfFormat::kOrc && type.type()->isDecimal()) {
+        return std::make_unique<DecimalColumnWriter>(
+            context, type, sequence, onRecordPosition);
+      }
```

Comment:
Let's throw if we encounter this in DWRF

Reply:
Added this check to throw for DWRF format. Thanks.

> VELOX_CHECK_EQ(format, DwrfFormat::kOrc, "Decimal is not supported for DWRF.");

Reply:
@Yuhta I notice the workflow failed with this check. Test like `TableScanTest, decimalDisableAggregationPushdown` relies on the IntegerColumnWriter to write decimal values to DWRF file, so we cannot throw for DWRF. How do you think? Thanks.

Reply:
workflow history: https://github.com/facebookincubator/velox/actions/runs/13421264854/job/37494224136